packages/babel-helpers/src/helpers/objectDestructuringEmpty.ts

```diff
@@ -0,0 +1,7 @@
+/* @minVersion 7.0.0-beta.0 */
+
+export default function _objectDestructuringEmpty<T>(
+  obj: T | null | undefined,
+): asserts obj is T {
+  if (obj == null) throw new TypeError("Cannot destructure " + obj);
+}
```

Comment:
`_objectDestructuringEmpty` is basically a `nullthrow` with a custom error message, so `asserts` is used.